src/profile/helpers/editData.ts

```diff
@@ -150,17 +182,254 @@ function pickSources(
   }
 }
 
+export function cloneAndMutateItem(
+  data: EditData,
+  overrides?: Partial<EditData>
+): EditData {
+  return {
+    ...data,
+    ...overrides,
+  };
+}
+
+export function findItemIndex(
+  allItems: EditData[],
+  idOrEditData: string | EditData
+): number {
+  const itemId =
+    typeof idOrEditData === 'string' ? idOrEditData : idOrEditData.id;
+  return allItems.findIndex(item => itemId === item.id);
+}
+
+function findItem(allItems: EditData[], id: string): EditData | undefined {
+  return allItems.find(item => id === item.id);
+}
+
+export function updateItemAndCloneList(
+  allItems: EditData[],
+  item: EditData,
+  newValue: EditDataValue,
+  saving?: SaveType
+): EditData[] {
+  const index = findItemIndex(allItems, item.id);
+  if (index < 0) {
+    throw new Error('Item not found in updateItemAndCloneList() ');
+  }
+  const updatedItem = cloneAndMutateItem(item, {
+    value: newValue,
+    saving: saving || undefined,
+  });
+  const newList = _.cloneDeep(allItems);
+  newList[index] = updatedItem;
+  return newList;
+}
+
+export function createNewProfileNode<T extends MultiItemProfileNode>(
```

Comment:
Did not find any usages outside of this file, so maybe the export is unnecessary? Though there's one similar looking function at myProfileMocking.ts line 211, maybe these could be combined?

Reply:
Yes, I merged them.